### DISCUSSION TITLE
Renaming default branch in GitHub causes error

### DISCUSSION THREAD

**Discussion by 0blu** [GitTools/GitVersion#3769] [Q&A] [answered]
**Describe the bug**
I was renaming my `master` branch to `stable` but that caused an error in GitVersion.
![image](https://github.com/GitTools/GitVersion/assets/7039020/4bfef961-3962-406b-9408-103e5e0e7282)

## Expected Behavior
To keep track of the version like it is `master`.

## Actual Behavior
`GitVersion.GitVersionException: No base versions determined on the current branch.`

## Steps to Reproduce
1) Create a repository on GitHub.
2) Do some commits and tag releases
3) Rename the main branch to something else
4) You should not be able to build the project anymore

## Environment
- Version Used: `6.0.0-beta.3`
- Operating System: Windows 10

[github_action_build_log.txt](https://github.com/GitTools/GitVersion/files/13341246/github_action_build_log.txt)

**Comment by asbjornu** [comment, marked as answer]
GitVersion has no idea what your `stable` branch is. You need to tell it that it is your `main` branch with the following [branch configuration](https://gitversion.net/docs/reference/configuration#branch-configuration):

```yaml
branches:
  main:
    regex: ^stable$
```
